src/components/sign-up/sign-up.jsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import "./sign-up.css";
+import endpointUrl from "../../config";
+export default () => {
+  let { token } = useParams();
+
+  const [user, setUser] = React.useState({
+    firstName: "",
+    surname: "",
+    email: "",
+    password: "",
+    confirmPassword: ""
+  });
+  const handleSubmit = e => {
+    const sendUserDetails = { ...user };
+    delete sendUserDetails.confirmPassword;
+    fetch(`${endpointUrl}/sign-up/${token}`, {
+      method: "POST",
+      body: JSON.stringify({ user: sendUserDetails }),
+      headers: {
+        "Content-Type": "application/json"
+      }
+    }).then(res => {
+      if (res.status === 302) {
+        window.location = "/";
+      }
+    });
+  };
+
+  const handleChange = ({ currentTarget: input }) => {
+    setUser({ ...user, [input.name]: input.value });
+  };
+  return (
+    <div className="sign-up-form-container">
+      <h1>sign up page</h1>
+
+      <div className="sign-up-form">
```

Comment:
Because your inputs are not in a `<form>`, you don't get the benefit of the validation. Eg: `required` and `type=email` don't do anything at the moment - your inputs need to be inside a `<form>` for these to work.